thirdparty/xatlas/xatlas.cpp

```diff
@@ -1955,7 +1955,10 @@ class Mesh
 
 		//make sure its not a degenerate
 		bool degenerate = distanceSquared(base[0], base[1]) < NV_EPSILON || distanceSquared(base[0], base[2]) < NV_EPSILON || distanceSquared(base[1], base[2]) < NV_EPSILON;
-		xaDebugAssert(!degenerate);
+		if (degenerate) {
```

Comment:
Please mark those Godot-specifix changes enclosed with `// -- GODOT start --` and `//-- GODOT end --`

Reply:
And include a .patch file with this change or merge in the other related patch.

Reply:
You can do e.g.:
```
cd thirdparty/xatlas/
git apply -R avoid-failing-on-bad-geometry.patch
git add xatlas.cpp 
git apply avoid-failing-on-bad-geometry.patch
// do your own changes
git diff xatlas.cpp > avoid-failing-on-bad-geometry.patch
git add xatlas.cpp
```
That should update the patch and leave your additional changes to commit.

Reply:
Will look into this this weekend.

Reply:
Instead of revising this patch, xatlas will be upgraded to the latest version.